### PR TITLE
Fix for UCP Listener created spark.port.maxRetries times

### PR DESCRIPTION
**Pull request by abellina**
Signed-off-by: Alessandro Bellina <abellina@nvidia.com>

Closes https://github.com/NVIDIA/spark-rapids/issues/2474

This bug shows when the UCP listener connection method is enabled (this is an internal config, and is disabled by default). It was just connecting up to all the retry attempts, instead of exiting once the listener could bind.

### CODE REVIEW

**Comment by abellina** [comment]
@petro-rudenko fyi

File: shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala [hunk 1]
```diff
@@ -180,16 +180,19 @@ class UCX(executor: BlockManagerId, rapidsConf: RapidsConf) extends AutoCloseabl
...
             listener = Option(worker.newListener(ucpListenerParams))
           } catch {
-            case ex: UcxException =>
-              logDebug(s"Failed to bind UcpListener on $sockAddress")
+            case _: UcxException =>
+              logDebug(s"Failed to bind UcpListener on $sockAddress. " +
+                s"Attempt ${attempt + 1} out of $maxRetries.")
               listener = None
           }
+          attempt += 1
```
Comment (jlowe) [inline shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala:195]: Nit: Move this to the top of the loop and we won't need to "fix it" in the log message.

**Review by jlowe — DISMISSED** [review]
Minor nit but otherwise lgtm.

File: shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala (hunk 1 above)
Comment (abellina) [reply shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala]: Made that change.

**Comment by jlowe** [comment]
build

**Review by petro-rudenko — APPROVED** [review]
Thanks

**Pull request merged**